request.go

```diff
@@ -51,7 +51,7 @@ func (r *callRequest) LogCompletion(res interface{}, err error) {
 func (r *callRequest) Reply(enc encoder, res interface{}, errArg interface{}) (err error) {
 	select {
 	case <-r.ctx.Done():
-		err = newCanceledError(r.Name(), r.SeqNo())
+		err = r.ctx.Err()
```

Comment:
Since you fix the hang below, you don't need this outer select -- the cancellation will be picked up by EncodeAndWrite or the second select below. So may as simplify it.